opensquirrel/squirrel_ir.py

```diff
@@ -79,8 +82,13 @@ class Statement(IRNode, ABC):
 
 
 class Gate(Statement, ABC):
-    arguments: Optional[Tuple[Expression, ...]] = None
+    # Note: two gates are considered equal even when their generators/arguments are different.
```

Comment:
In the gate set there might be different ways to create the same "gate semantic"
Classic example:
`x q[i]` is the same as `rx q[i], pi` up to irrelevant global phase

That's why I chose to compare only the semantic/matrix up to global phase, and not the way this semantic is achieved

Reply:
Here is where my interpretation of argument/parameter has failed me 😅! What I meant was: how could equal gates have different __qubit parameters__? But if 'arguments' also includes non-qubit parameters, I see how.

Reply:
If they operate on different qubits they are not equal
Since the qubits are part of the semantic
You are correct that the qubit is also specified in the arguments
Equal gates CANNOT have different qubit parameters (except if the gate is ill-defined and has a qubit parameter that it doesn't use...)
Between the parentheses means: if I define
```python
@named_gate
def stupid_x(q: Qubit, useless_q: Qubit):
  return BlochSphereRotation(q, pi, pi/2) # Notice that useless_q is not used...
```
This raises a lot of questions actually
For instance, if we define identity, is `id q[0]` equal to `id q[1]`?

Reply:
If I define `stupid_x` that way, then `stupid_x(Qubit(0), Qubit(1))` will be "equal" for this `__eq__` operator to `stupid_x(Qubit(0), Qubit(2))` even though they have different qubit parameters
Since the `useless_q` doesn't appear in the semantic and is therefore not checked by the `__eq__` of `BlochSphereRotation`



I think this concept of "generator" and "arguments" is not so clear
Should improve it and write about it.

So what is clear is that a gate is a semantic. Those extra parameters explain HOW this semantic was obtained.

Reply:
Clear! Thanks for the explanation.

Reply:
https://github.com/QuTech-Delft/OpenSquirrel/issues/92